File: class-diagram-generator/sample-projects/gradle-single-jdk17/src/main/java/com/example/catalog/repository/ProductRepository.java
package com.example.catalog.repository;

import com.example.catalog.model.Product;
import java.util.List;
import java.util.Optional;

/**
 * 상품 저장소의 조회 계약을 정의한다.
 */
public interface ProductRepository {
    /**
     * SKU 기준으로 단일 상품을 조회한다.
     *
     * @param sku 상품 식별자
     * @return 조회 결과
     */
    Optional<Product> findBySku(String sku);

    /**
     * 키워드로 상품을 검색한다.
     *
     * @param keyword 검색 키워드
     * @return 검색 결과 목록
     */
    List<Product> findByKeyword(String keyword);

    /**
     * 전체 상품을 조회한다.
     *
     * @return 전체 상품 목록
     */
    List<Product> findAll();
}

File: class-diagram-generator/sample-projects/maven-single-kotlin-jdk21/src/main/kotlin/com/example/catalog/repository/ProductRepository.kt
package com.example.catalog.repository

import com.example.catalog.model.ProductStatus
import com.example.catalog.model.ProductSummary

interface ProductRepository {
    fun findAll(): List<ProductSummary>
}

class InMemoryProductRepository : ProductRepository {
    private val data = listOf(
        ProductSummary(1L, "Kotlin in Action", "cat-001", ProductStatus.ACTIVE),
        ProductSummary(2L, "Spring Recipes", "cat-002", ProductStatus.OUT_OF_STOCK),
        ProductSummary(3L, "Legacy Java", "cat-003", ProductStatus.INACTIVE)
    )

    override fun findAll(): List<ProductSummary> = data

    companion object {
        const val SOURCE = "in-memory"
    }
}
